lib/vagrant/synced_folder.rb

```diff
@@ -0,0 +1,16 @@
+require "vagrant/capability_host"
+
+module Vagrant
+  # This class handles host-OS specific interactions. It is responsible for
+  # detecting the proper host OS implementation and delegating capabilities
+  # to plugins.
+  #
+  # See {Guest} for more information on capabilities.
+  class SyncedFolder
```

Comment:
might be able to pop this into the [synced folder plugin bit](https://github.com/hashicorp/vagrant/blob/master/lib/vagrant/plugin/v2/synced_folder.rb)